src/Avalonia.Controls/TreeView.cs

```diff
@@ -52,6 +52,18 @@ public class TreeView : ItemsControl, ICustomKeyboardNavigation
         /// </summary>
         public static readonly StyledProperty<SelectionMode> SelectionModeProperty =
             ListBox.SelectionModeProperty.AddOwner<TreeView>();
+        
+        /// <summary>
+        /// Defines the <see cref="Expanded"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> ExpandedEvent =
+            RoutedEvent.Register<TreeView, RoutedEventArgs>(nameof(Expanded), RoutingStrategies.Bubble);
+        
+        /// <summary>
+        /// Defines the <see cref="Collapsed"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> CollapsedEvent =
+            RoutedEvent.Register<TreeView, RoutedEventArgs>(nameof(Collapsed), RoutingStrategies.Bubble);
```

Comment:
Sorry, I haven't noticed at first, but why does TreeView has these events? And not TreeViewItem? Makes more sense to be on the item.

It's still possible to handle these events on the TreeView or any other  higher-position control, since this event bubbles:
```csharp
treeView.AddHandler(TreeViewItem.CollapsedEvent, Handler);
```

Also with 11.2, it's possible to subscribe to any routed event:
```xaml
<TreeView TreeViewItem.CollapsedEvent="Handler" ItemsSource="{Binding Source}" />
```

Reply:
I made the design from user prototype (directly add class handler in TreeView level in xaml) but yes, with the new RoutedEvent update we can add with your second sample. Can I assume this PR will be merged in 11.2 lifecycle?

Reply:
Most likely, yes, it will go into 11.2, if merged this week.